Forwarded: not-needed
 Breaks regulatory requirements (edge channels) and should not be used

--- a/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c
+++ b/drivers/net/wireless/ath/ath9k/ar9003_eeprom.c
@@ -5151,6 +5151,11 @@ static void ar9003_hw_set_power_per_rate
 		       __func__, ctlMode, pCtlMode[ctlMode], twiceMaxEdgePower,
 			scaledPower, minCtlPower);
 
+		minCtlPower = scaledPower;
+		/* TODO remove: printk("%s(): [modified] ctlMode %d pCtlMode %d 2xMaxEdge %d sP %d minCtlPwr %d\n",
+		       __func__, ctlMode, pCtlMode[ctlMode], twiceMaxEdgePower,
+			scaledPower, minCtlPower); */
+
 		ath_dbg(common, REGULATORY,
 			"SEL-Min ctlMode %d pCtlMode %d 2xMaxEdge %d sP %d minCtlPwr %d\n",
 			ctlMode, pCtlMode[ctlMode], twiceMaxEdgePower,
--- a/drivers/net/wireless/ath/ath9k/eeprom_4k.c
+++ b/drivers/net/wireless/ath/ath9k/eeprom_4k.c
@@ -566,6 +566,12 @@ static void ath9k_hw_set_4k_power_per_ra
 		       __func__, ctlMode, pCtlMode[ctlMode], twiceMaxEdgePower,
 			scaledPower, minCtlPower);
 
+		scaledPower = max((u16)0, powerLimit);
+		minCtlPower = scaledPower;
+		/* TODO remove: printk("%s() [modified]: ctlMode %d pCtlMode %d 2xMaxEdge %d sP %d minCtlPwr %d\n",
+		       __func__, ctlMode, pCtlMode[ctlMode], twiceMaxEdgePower,
+			scaledPower, minCtlPower); */
+
 		switch (pCtlMode[ctlMode]) {
 		case CTL_11B:
 			for (i = 0; i < ARRAY_SIZE(targetPowerCck.tPow2x); i++) {
